pkg/apis/core/validation/validation.go

```diff
@@ -136,7 +136,7 @@ func ValidateDNS1123Label(value string, fldPath *field.Path) field.ErrorList {
 }
 
 // ValidateQualifiedName validates if name is what Kubernetes calls a "qualified name".
-func ValidateQualifiedName(value string, fldPath *field.Path) field.ErrorList {
+func ValidateQualifiedName[T ~string](value T, fldPath *field.Path) field.ErrorList {
```

Comment:
I don't think we should do this one - the original type is not relevant.  Let callers conver this to string.